ambari-server/src/main/python/ambari_server/serverSetup.py

```diff
@@ -80,7 +80,7 @@
 JDK_PROMPT = "[{0}] {1}\n"
 JDK_VALID_CHOICES = "^[{0}{1:d}]$"
 
-JDK_VERSION_CHECK_CMD = """{0} -version 2>&1 | grep -i version | sed 's/.*version ".*\.\(.*\)\..*"/\\1/; 1q' 2>&1"""
+JDK_VERSION_CHECK_CMD = """{0} -version 2>&1 | grep -i version 2>&1"""
```

Comment:
Looks like a corresponding change is needed in the unit test.  These mock return values should include the full line of output with version instead of only the number:

https://github.com/apache/ambari/blob/44436e1d5e0d28e07285c7b506cf596748ce85fc/ambari-server/src/test/python/TestAmbariServer.py#L3153

https://github.com/apache/ambari/blob/44436e1d5e0d28e07285c7b506cf596748ce85fc/ambari-server/src/test/python/TestAmbariServer.py#L3162

eg.

```
    p.communicate.return_value = ('java version "1.7.0_191"', None)
```

and

```
    p.communicate.return_value = ('openjdk version "1.8.0_161"', None)
```